`azurerm_postgresql_flexible_server` - doc enhancement (#19673)
hashicorp/terraform-provider-azurerm: website/docs/r/postgresql_flexible_server.html.markdown

* `tenant_id` - (Optional) The Tenant ID of the Azure Active Directory which is used by the Active Directory authentication. `active_directory_auth_enabled` must be set to `true`.

> -> **Note:** Setting `active_directory_auth_enabled` to `true` needs to set up the Service Principal of Postgres Flexible Server first. Details could be found on [this document](https://learn.microsoft.com/en-us/azure/postgresql/flexible-server/how-to-configure-sign-in-azure-ad-authentication).

Reviewer: ```suggestion
-> **Note:** Setting `active_directory_auth_enabled` to `true` requires a Service Principal for the Postgres Flexible Server. For more details see [this document](https://learn.microsoft.com/en-us/azure/postgresql/flexible-server/how-to-configure-sign-in-azure-ad-authentication).
```
